core/src/main/java/io/confluent/kafka/schemaregistry/rest/SchemaRegistryRestApplication.java

```diff
@@ -122,6 +122,14 @@ public void setupResources(Configurable<?> config, SchemaRegistryConfig schemaRe
         System.exit(1);
       }
     }
+
+    // Initiate leader election only after all resources have been set up.
```

Comment:
I think your change will address the issue. The only callout is the leader election breaks the single responsibility of this particular method `setupResources`. A more thorough way is to define a method that will be called after `setupResources` in the base class, and override it.

Reply:
@Clarence97 Apart from @ericwush 's point about single responsibility, it looks like we have another reason to not add this logic here. Looking at the code, I see that the SR instance's http(s) endpoint is actually not ready to listen to requests event at this point. The listeners binding to the port (and a bunch of other server initialization) happens in [server.start()](https://github.com/confluentinc/schema-registry/blob/3c1d948947301213242da604be32adf42c862449/core/src/main/java/io/confluent/kafka/schemaregistry/rest/SchemaRegistryMain.java#L44) - and I did a quick test to verify that the server is ready to receive requests only after this method returns. So we need to move leader election further to fully address the issue. 
A possible fix is to add `initLeaderElection` to `SchemaRegistryRestApplication` (can pick a more generic name to accommodate any other future logic that might need to go in here) which in turn calls `KafkaSchemaRegistry::initLeaderElection` and call it after `server.start()` in `SchemaRegistryMain`.

Reply:
Apart from @ericwush 's comment about single responsibility, it looks like we have another reason not to add this logic here. From the code, I see that the SR instance is not able to receive requests even after `setupResources` is complete. The server is fully ready only after [server.start()](https://github.com/confluentinc/schema-registry/blob/3c1d948947301213242da604be32adf42c862449/core/src/main/java/io/confluent/kafka/schemaregistry/rest/SchemaRegistryMain.java#L44) returns, which is where listeners binding to ports and a bunch of other initialization happens. Initiating leader election right after `setupResources` still leaves a tiny window where requests may fail, so we should consider doing it after `server.start()`.

Reply:
Agree this seems to be a better place to call `initLeaderElection` and would also avoid unnecessary changes to rest-utils's `Application` class.

Reply:
Thanks @ericwush , @suj-krishnan and @sanjay-awatramani ! Agreed. Moving the leader election to be after everything is set up seems to be the best option.